### PR TITLE
MM-60171: Using a generic function to allocate values

### PR DESCRIPTION
**Pull request by agnivade**
We sprinkle a bit of generic magic to refactor a lot
of duplicate code.

To avoid exposing unnecessary code, I duplicated the function
twice. But let me know if you have strong opinions about this.

https://mattermost.atlassian.net/browse/MM-60171

```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
This is branched off of https://github.com/mattermost/mattermost/pull/28143. Not to be merged until that is merged.

**Review by agarciamontoro — APPROVED** [review]
Oh, neat, love to see PRs using generics <3 I'm not very concerned with the duplication you mention, but exposing the function would not be the end of the world either. Up to you, don't have a strong opinion on that.
What I don't like 100% is the name of the function, but I don't have a good enough alternative for that either. I was thinking about something like `allocateTargetMap`. The thing is that `getCacheTargets` sounds like you are actually *getting* some values from somewhere. Thoughts?

**Comment by agnivade** [comment]
How about `allocateCacheTargets`?

**Comment by agarciamontoro** [comment]
> How about `allocateCacheTargets`?

That sounds good to me!

File: server/channels/app/platform/utils.go [hunk 1]
```diff
@@ -20,3 +20,14 @@ func maxInt(a, b int) int {
...
 	return b
 }
+
+// getCacheTargets is used to fill target value types
+// for getting items from cache.
+func getCacheTargets[T any](l int) []any {
+       toPass := make([]any, 0, l)
+       for i := 0; i < l; i++ {
+               var target T
+               toPass = append(toPass, &target)
```
Comment (streamer45) [inline server/channels/app/platform/utils.go:30]: Apologies if this is a known gotcha, but can't we simply append `new(T)`?

Comment (agnivade) [reply server/channels/app/platform/utils.go:30]: Should be possible. I'll confirm.

**Review by streamer45 — APPROVED** [review]
Cool!

**Pull request merged**